core/src/openid.cpp

```diff
@@ -98,16 +98,79 @@ namespace irods::http::openid
 	{
 		namespace logging = irods::http::log;
 
-		body_arguments args{{"token", _bearer_token}, {"token_type_hint", "access_token"}};
+		const body_arguments args{{"token", _bearer_token}, {"token_type_hint", "access_token"}};
 
 		auto json_res{hit_introspection_endpoint(url_encode_body(args))};
 
-		// Validate access token
-		if (!json_res.at("active").get<bool>()) {
+		// Check that the access token is active.
+		// The exact meaninng of an active state may vary between OpenID Providers,
+		// so we should assert the other attributes when available.
+		if (const auto active_iter{json_res.find("active")};
+		    active_iter == std::end(json_res) || !active_iter->get<bool>()) {
 			logging::warn("{}: Access token is invalid or expired.", __func__);
 			return std::nullopt;
 		}
```

Comment:
Firstly...
```suggestion
		// Check that the access token is active.
		// The exact meaning of an active state may vary between OpenID Providers,
		// so we should assert the other attributes when available.
		if (const auto active_iter{json_res.find("active")};
		    active_iter == std::end(json_res) || !active_iter->get<bool>()) {
			logging::warn("{}: Access token is invalid or expired.", __func__);
			return std::nullopt;
		}
```

Secondly... Does this mean that we do *not* want to `return` before checking the "other attributes"? The comment suggests to me that an OpenID Provider not providing an "active" attribute or if the "active" attribute is `false` does not necessarily indicate that the token is *not* active. Am I reading that wrong?

Reply:
A response with the `active` member set to `false` means that the token is completely invalid (e.g., expired, not recognized). Additionally, no other members are returned in this case.

When a response has `active` as `true`, this means that the provided access token is valid by some measure that the OpenID Provider specifies as active. There are general guarantees, but the specifics are implementation defined.

Quote from the RFC:
> The specifics of a token’s "active" state will vary depending on the implementation of the authorization server and the information it keeps about its tokens, but a "true" value return for the "active" property will generally indicate that a given token has been issued by this authorization server, has not been revoked by the resource owner, and is within its given time window of validity (e.g., after its issuance time and before its expiration time).

Link to response info: https://www.rfc-editor.org/rfc/rfc7662.html#section-2.2

Hopefully, that explains the weird phrasing.

Reply:
Okay, I see. So, as a general matter, the token *at least* must be "active" == `true`. Beyond that, some providers will have additional checks that must be considered below. The way it's phrased now made me think that some providers might not set "active" but *would* set the other attributes.

Reply:
So, something closer to this?
`// The exact meaning of the state member being true may vary between OpenID Providers...`

Reply:
Added more words.